src/MatrixTwitter.js

```diff
@@ -0,0 +1,1040 @@
+        intent.setDisplayName(profile.name + " (@" + profile.screen_name + ")");
+      }
+      else {
+        log.warn("Twitter", "Tried to preform a user display name update with a null profile.");
+      }
+    }
+
+    if(update_avatar) {
+      if(profile == null || profile.profile_image_url_https == null) {
+        log.warn("Twitter", "Tried to preform a user avatar update with a null profile.");
+        return;
+      }
+      util.uploadContentFromUrl(this._bridge, profile.profile_image_url_https, intent).then((uri) =>{
+        return intent.setAvatarUrl(uri);
+      }).catch(err => {
+        log.error(
+            'Twitter',
+            "Couldn't set new avatar for @%s because of %s",
+            profile.screen_name,
+            err
+          );
+      });
+    }
+    this.storage.cache_user_profile(profile.id, profile.screen_name, profile, ts);
+  });
+}
+
+MatrixTwitter.prototype._create_twitter_client = function (creds) {
+  var ts = new Date().getTime();
+  var client = new Twitter({
+    consumer_key: this.app_auth.consumer_key,
+    consumer_secret: this.app_auth.consumer_secret,
+    access_token_key: creds.access_token,
+    access_token_secret: creds.access_token_secret
+  });
+  client.last_auth = ts;
+  return client;
+}
+
+MatrixTwitter.prototype._get_twitter_client = function (sender) {
+  //Check if we have the account in the cache
+  return this.storage.get_twitter_account(sender).then((creds) => {
+    return new Promise( (resolve, reject) => {
+      if(creds == null) {
+        reject("No twitter account linked.");
+        return;
+      }
+
+      var ts = new Date().getTime();
+      var id = creds.user_id;
+      var client;
+      if(this.tclients.has(id)) {
+        client = this.tclients[id];
+        if(ts - client.last_auth < TWITTER_CLIENT_INTERVAL_MS) {
+          resolve(client);
+          return;
+        }
+
+        log.info("Twitter", "Credentials for %s need to be reevaluated.", sender);
+        client.get("account/verify_credentials", (error, profile) => {
+          if(error) {
+            log.info("Twitter", "Credentials for " + id + " are no longer valid.");
+            log.error("Twitter", error);
+            delete this.tclients[id];//Invalidate it
+            resolve(this._get_twitter_client(sender));
+          }
+          client.profile = profile;
+          this._update_user_timeline_profile(profile);
+          resolve(client);
+        });
+      }
+      else {
+        client = this._create_twitter_client(creds);
+        this.tclients[id] = client;
+        client.get("account/verify_credentials", (error, profile) => {
+          if(error) {
+            delete this.tclients[id];//Invalidate it
+            log.error(
+              "Twitter",
+              "We couldn't authenticate with the supplied access token for %s. Look into this.",
+              id
+            );
+            reject("Twitter account could not be authenticated.");
```

Comment:
You should really be logging the `error` *somewhere* so you can at least debug stuff!